docs/devops.rst

```diff
@@ -0,0 +1,91 @@
+Devops
+===========
+
+Azure Pipeline
+--------------
+
+``anonlink-client`` is automatically built and tested using Azure Pipeline for Windows environment, in the project `Anonlink <https://dev.azure.com/data61/Anonlink>`_
+
+Two pipelines are available:
+- `Build pipeline <https://dev.azure.com/data61/Anonlink/_build?definitionId=6>`_,
+- `Release pipeline <https://dev.azure.com/data61/Anonlink/_release?definitionId=5>`_.
+
+
+Build Pipeline
+~~~~~~~~~~~~~~
+
+The build pipeline is described by the script `azurePipeline.yml`.
+
+There are 3 top level stages in the build pipeline:
+
+- *Static Checks* - runs `mypy` typechecking over the codebase. Also adds a Azure DevOps tag `"Automated"`
+  if the build was triggered by a Git tag.
+- *Unit tests* - A template expands out into a number of builds and tests for different
+  version of python and system architecture.
+- *Packaging* - Pulls together the created files into a single release artifact.
+
+
+The *Build & Test* job does:
+
+- install the requirements,
+- package ``anonlink-client``,
+- run tests as described in the following table,
+- publish the test results,
+- publish the code coverage (on Azure and codecov),
+- publish the artifacts from the build using ``Python 3.7`` (i.e. the wheel, the sdist `tar.gz` and an exe for x86 and x64).
+
+The build pipeline requires one environment variable provided by Azure environment:
+
+- `CODECOV_TOKEN` which is used to publish the coverage to codecov.
+
+Description of what is tested:
+
+==================   ====================  ===============  ===========  ===================  =========
+Python Version       Operating System      Standard pytest  INLCUDE_CLI  TEST_ENTITY_SERVICE  Notebooks
+==================   ====================  ===============  ===========  ===================  =========
+pypy2                ubuntu-18.04          Yes              No           No                   No
+pypy3                ubuntu-18.04          Yes              No           No                   No
```

Comment:
I don't see pypy in the Azure pipeline. Is this up to date?

Reply:
Azure pipeline does not include pypy2 & 3 so I just removed them